Src/Support/Google.Apis.Core/Requests/Parameters/ParameterValidator.cs

```diff
@@ -29,22 +29,42 @@ public static class ParameterValidator
     {
         /// <summary>Validates a parameter value against the methods regex.</summary>
         [VisibleForTestOnly]
-        public static bool ValidateRegex(IParameter param, string paramValue)
+        public static bool ValidateRegex(IParameter param, string paramValue, out string error)
         {
-            return string.IsNullOrEmpty(param.Pattern) || new Regex(param.Pattern).IsMatch(paramValue);
+            if (string.IsNullOrEmpty(param.Pattern) || new Regex(param.Pattern).IsMatch(paramValue))
+            {
+                error = null;
+                return true;
+            }
+            else
+            {
+                error = string.Format("The value did not match the regex {0}", param.Pattern);
```

Comment:
I'd just use an interpolated string literal here:

```csharp
error = $"The value did not match the regular expression {param.Pattern}";
```

(I think it's probably slightly more friendly to expand regex to regular expression.)